### ISSUE TITLE
Feature - Assembly 3 - delete or reuse an unused Element in an Assembly ?

### ISSUE THREAD

**Issue by thx1111** [realthunder/FreeCAD#888]
When testing and deleting Constraints in an Assembly, a new set of Elements are generated for each Constraint.  When the Constraint is removed, the related Elements still remain, now unused.  This process of deleting Constraints can produce a long list of unused and unnecessary Elements, which may be deleted manually after reviewing the remaining active Constraints.

Does it make sense to automatically remove the Elements created for a Constraint when the Constraint is deleted?  Or instead, to automatically scan the list of Elements for an unused equivalent Element, to be reused?

**Comment by wsteffe** [comment]
I also agree that it would be better to automatically delete unused elements after a constraint i deleted.
And I also would like that after deleting a Link in Assembly/Parts, all constraints referring to it (trough an associated element) are automaltically removed (similarly to what happens deleting a constraint in Sketch WB).
In the current version of Assembly3 when a Link is deleted in Assembly/Parts the document gets broken with several errors reported. In my opinion this behaviour is very bad.
